src/models/blockchain/midnight/transaction.rs

```diff
@@ -76,6 +141,64 @@ impl From<MidnightRpcTransaction> for Transaction {
 	}
 }
 
+impl From<Transaction> for MidnightRpcTransaction {
+	fn from(tx: Transaction) -> Self {
+		tx.0
+	}
+}
+
+impl<P: Proofish<D>, D: DB> From<ContractAction<P, D>> for Operation {
+	fn from(action: ContractAction<P, D>) -> Self {
+		match action {
+			ContractAction::Call(call) => Operation::Call {
+				address: call.address.0 .0.encode_hex(),
+				entry_point: String::from_utf8_lossy(&call.entry_point.0).to_string(),
+			},
+			ContractAction::Deploy(deploy) => Operation::Deploy {
+				address: deploy.address().0 .0.encode_hex(),
+			},
+			ContractAction::Maintain(update) => Operation::Maintain {
+				address: update.address.0 .0.encode_hex(),
+			},
+		}
+	}
+}
+
+impl<D: DB>
+	TryFrom<(
+		Transaction,
+		Option<MidnightNodeTransaction<Proof, D>>,
+		&Vec<ChainConfiguration>,
+	)> for Transaction
+{
+	type Error = anyhow::Error;
+
+	fn try_from(
+		(block_tx, ledger_tx, chain_configurations): (
+			Transaction,
+			Option<MidnightNodeTransaction<Proof, D>>,
+			&Vec<ChainConfiguration>,
+		),
+	) -> Result<Self, Self::Error> {
+		// Check if chain_configuration has viewing keys and decrypt the transaction's coins
+		for chain_configuration in chain_configurations {
+			if let Some(midnight) = &chain_configuration.midnight {
+				for viewing_key in &midnight.viewing_keys {
+					if let SecretValue::Plain(secret) = viewing_key {
+						let viewing_key_str = secret.as_str();
+						if let Some(ref ledger_tx) = ledger_tx {
+							// TODO: Do something with the coins...
```

Comment:
This is intentional. We wanted this logic in place already. We will extend this in the future.